### PR TITLE
Prevent `NoMethodError` in extract_value when specifying non-existent keys

### PR DESCRIPTION
**Pull request by euglena1215**
### Motivation / Background

Introduced in https://github.com/rails/rails/pull/49042, the method `ActionController::Parameters#extract_value` promises to replace utility methods that were previously defined as private methods in controllers.
However, it currently throws a `NoMethodError` when passed a non-existent key.

`params` is dependent on client requests and is thus beyond the application's control.
Rather than throwing a `NoMethodError`, it would be more convenient for the method to return `nil`.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

File: actionpack/lib/action_controller/metal/strong_parameters.rb [hunk 1]
```diff
@@ -931,7 +931,9 @@ def deep_dup
     #   params = ActionController::Parameters.new(id: "1_123", tags: "ruby,rails")
     #   params.extract_value(:id) # => ["1", "123"]
     #   params.extract_value(:tags, delimiter: ",") # => ["ruby", "rails"]
+    #   params.extract_value(:non_existent_key) # => []
     def extract_value(key, delimiter: "_")
+      return [] if @parameters[key].nil?
       @parameters[key].split(delimiter)
```
Comment (byroot) [inline actionpack/lib/action_controller/metal/strong_parameters.rb:937]:
```suggestion
      @parameters.fetch(key) { return [] }.split(delimiter)
```

Might as well avoid hashing (and casting) the key twice.

Comment (euglena1215) [reply actionpack/lib/action_controller/metal/strong_parameters.rb:937]:
```suggestion
      @parameters.fetch(key, "").split(delimiter)
```
Although the split method is called redundantly, this is simpler. What do you think?
This method also avoids hashing the key twice.

**Comment by byroot** [comment]
Please squash your commits.

**Comment by byroot** [comment]
I'll wait for @nvasilevski 's opinion, but LGTM.

**Comment by smoburgower** [comment]
ICU Sent from my iPhoneOn Aug 31, 2023, at 03:01, Shintani Teppei ***@***.***> wrote:﻿
@euglena1215 commented on this pull request.



In actionpack/lib/action_controller/metal/strong_parameters.rb:
> +      return [] if @parameters[key].nil?
       @parameters[key].split(delimiter)

⬇️ Suggested change
-      return [] if @parameters[key].nil?
-      @parameters[key].split(delimiter)
+      @parameters.fetch(key, "").split(delimiter)


Although the split method is called redundantly, this is simpler. What do you think?
This method also avoids hashing the key twice.

**Review by nvasilevski — APPROVED** [review]
👍 
Thanks, this makes total sense

**Comment by matthewd** [comment]
To me it feels like this should perhaps be nil 🤔

**Comment by euglena1215** [comment]
OK. It make sense. So I'll fix it to return `nil`.

**Pull request merged**